### PR TITLE
Add otNewIp6Message API

### PR DESCRIPTION
**Pull request by nibanks**
Adds a new API, fixes the otAppendMessage API (it's actually already returning the error), and updates NCP to use the new API.

### CODE REVIEW

File: include/openthread.h [hunk 1]
```diff
@@ -1669,7 +1670,7 @@ ThreadError otSetMessageOffset(otMessage aMessage, uint16_t aOffset);
  * @sa otReadMessage
  * @sa otWriteMessage
  */
-int otAppendMessage(otMessage aMessage, const void *aBuf, uint16_t aLength);
+ThreadError otAppendMessage(otMessage aMessage, const void *aBuf, uint16_t aLength);
```
Comment (darconeous) [inline include/openthread.h]: As a quick aside not related to these changes, we might want to consider renaming the message functions to all have the prefix `otMessage*()`. This makes it more clear that the subject of the method is an `otMessage` object. `otAppendMessage()` reads like we are appending an `otMessage` onto something, which is confusing.

Again, this isn't a comment related to this commit, just a note for future reference.

**Comment by darconeous** [comment]
LGTM 👍 

But I bet you are in merge hell right now, considering your pending changes for #473. Considering that there will be merge conflicts no matter what order things get merged, is there any particular commit you would prefer merged first?

**Comment by nibanks** [comment]
Lol... I have kind of been in that state for the last month. :) I may spend as much as 2 hours a day doing merges and making sure everything builds. I don't really care what you merge first. I can easily merge the other with master and update the PR.

**Comment by abtink** [comment]
LGTM :+1: 

More of an observation for longer-term changes:
I noticed that we are still using `Message` pointers in the `NcpBase` and also they are used in `NcpFrameBuffer` (we also need `MessageQueue` and some of its method in ncp buffer). Basically in some places we use the `otMessage` and in some places we convert it back to `Message` and use the class methods.  I think we should harmonize this in longer-term. 

One approach is that all the relevant APIs are exposed through C `otMessage_<fn>` APIs to enable us to remove any dependency to `common/message.hpp` in Ncp classes.

**Comment by jwhui** [comment]
LGTM 👍

File: src/ncp/ncp_base.cpp [hunk 2]
```diff
@@ -3174,8 +3173,8 @@ ThreadError NcpBase::SetPropertyHandler_STREAM_NET_INSECURE(uint8_t header, spin
         (void)meta_ptr;
         (void)meta_len;
         (void)parsedLength;
-
-        errorCode = message->Append(frame_ptr, static_cast<uint16_t>(frame_len));
+
```
Comment (mszczodrak) [inline src/ncp/ncp_base.cpp]: extra space

File: src/ncp/ncp_base.cpp [hunk 3]
```diff
@@ -3245,8 +3243,8 @@ ThreadError NcpBase::SetPropertyHandler_STREAM_NET(uint8_t header, spinel_prop_k
...
```
Comment (mszczodrak) [inline src/ncp/ncp_base.cpp]: extra space

**Pull request merged**